modules/serverless-typescript-function-return-values.adoc

```diff
@@ -0,0 +1,74 @@
+[id="serverless-typescript-function-return-values_{context}"]
+= TypeScript function return values
+
+Functions can return any valid JavaScript type or can have no return value. When a function has no return value specified, and no failure is indicated, the caller receives a `204 No Content` response.
+
+Functions can also return a CloudEvent or a `Message` object in order to push events into the Knative Eventing system. In this case, the developer is not required to understand or implement the CloudEvent messaging specification. Headers and other relevant information from the returned values are extracted and sent with the response.
+
+.Example
+[source,javascript]
+----
+function handle(context: Context, cloudevent?: CloudEvent): CloudEvent {
+  // process customer and return a new CloudEvent
+  const customer = cloudevent.data;
+  return new CloudEvent({
+    source: 'customer.processor',
+    type: 'customer.processed'
+  })
+}
+----
+
+[id="serverless-typescript-function-return-values-headers_{context}"]
+== Returning headers
+
+You can set a response header by adding a `headers` property to the `return` object. These headers are extracted and sent with the response to the caller.
+
+.Example response header
+[source,javascript]
+----
+function handle(context: Context, cloudevent?: CloudEvent): Record<string, string> {
+  // process customer and return custom headers
+  // the response will be '204 No content'
+  const customer = cloudevent.data;
+  return { headers: { "customer-id": customer.id } };
+}
+----
+
+[id="serverless-typescript-function-return-values-status-codes_{context}"]
+== Returning status codes
+
+You can set a status code that is returned to the caller by adding a `statusCode` property to the `return` object:
+
+.Example status code
+[source,javascript]
+----
+function handle(context: Context, cloudevent?: CloudEvent): Record<string, string> {
```

Comment:
Here is a suggestion for this example. Got this working with expected returned status code using the below. /cc @lance 
```
export function handle(context: Context, cloudevent?: CloudEvent): Record<string, any> {
  // process customer
  const customer = cloudevent.data as Record<string, any>;
  if (customer.restricted) {
    return {
      statusCode: 451
    }
  }
  // business logic, then
  return {
    statusCode: 240
  }
}
```

Reply:
lgtm